File: Fleet.Api/Features/Ships/DTOs/OnShipUnloadedEvent.cs
using Fleet.Api.Entities;

namespace Fleet.Api.Features.Ships.DTOs;

public class OnShipUnloadedEvent
{
    public int ShipId { get; set; }
    public UnloadedContainer UnloadedContainer { get; set; }

    public OnShipUnloadedEvent(int shipId, Container container)
    {
        ShipId = shipId;
        UnloadedContainer = new UnloadedContainer(container);
    } 
}

public class UnloadedContainer
{
    public int Id { get; set; }
    public string Name { get; set; }
    
    
    public UnloadedContainer(Container container)
    {
        Id = container.Id;
        Name = container.Name;
    } 
}

File: Fleet.Client/src/app/_models/onShipUnloadedEvent.ts
import {Container} from "./container";

export interface OnShipUnloadedEvent {
  shipId: number;
  unloadedContainer: Container;
}

export function emptyShipUnloadedEvent(): OnShipUnloadedEvent{
  return { shipId: -1, unloadedContainer: { id: -1, name: '' } }
}
